src/condor_utils/file_transfer.cpp

```diff
@@ -4141,7 +4152,47 @@ FileTransfer::DoUpload(filesize_t *total_bytes_ptr, ReliSock *s)
 		}
 	}
 
+
+	//
+	// Remove file entries which result in duplicates at the destination.
+	// This is no longer necessary for correctness (because I changed
+	// FileTransferItem::operator < to group rather than sort), but it's
+	// still more efficient.  Some of these duplicates will have been
+	// generated internally, but not all.  It's more efficient to remove
+	// them all here (especially when the duplicate would only have been
+	// generated by expanding a directory).  We remove duplicate directory
+	// creation entries when we create the extra entries because for those,
+	// we want to preserve the earlier one, rather than the later one.
+	//
+	std::set<std::string> names;
+	for( auto iter = filelist.rbegin(); iter != filelist.rend(); ++iter ) {
+		auto & item = * iter;
+
+		if( item.isSrcUrl() ) { continue; }
+		if( item.isDestUrl() ) { continue; }
```

Comment:
Why not do duplicate detection on destination URLs as well?

Reply:
Why not do duplicate detection on source URLs?  Because neither has anything to do with the problem I was trying to solve.  (As far as I know, the HTCondor code never introduces URLs at all, much less duplicate ones.)

Reply:
What problem were you trying to solve then?  The comment says:

> Remove file entries which result in duplicates at the destination. This is no longer necessary for correctness [...], but it's still more efficient

Per the comment, this was done to improve efficiency, no?

Reply:
As per the ticket, the problem I was trying to solve was HTCondor's inability to resume certain self-checkpointing jobs after eviction, which had nothing to do with URLs.
Further, the destination URL can do literally whatever it wants with the file it's sent; for example, `foo = scp://my_server/my_output_directory; bar => scp://my_server/my_output_directory` has duplicate destination URLs but is not a duplicate transfer.